pkg/gce-pd-csi-driver/controller.go

```diff
@@ -921,6 +961,14 @@ func (gceCS *GCEControllerServer) createPDSnapshot(ctx context.Context, project
 			if gce.IsGCEError(err, "notFound") {
 				return nil, status.Errorf(codes.NotFound, "Could not find volume with ID %v: %v", volKey.String(), err.Error())
 			}
+			// Emit metric for error
+			if strings.Contains(sourceDisk.GetPDType(), "hyperdisk") {
+				gceCS.Metrics.RecordOperationErrorMetrics(hyperdiskDriverName, "/csi.v1.Controller/CreateSnapshot", err, sourceDisk.GetPDType())
+			}
+
+			if strings.Contains(sourceDisk.GetPDType(), "pd") {
+				gceCS.Metrics.RecordOperationErrorMetrics(pdcsiDriverName, "/csi.v1.Controller/CreateSnapshot", err, sourceDisk.GetPDType())
+			}
```

Comment:
For hyperdisk we should also emit errors related to if the snapshot is ready: gceCS.validateExistingSnapshot